Source/Plugins/Tilemaps/Core/Tilemaps/TilemapFloodFill.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+
+namespace Duality.Plugins.Tilemaps
+{
+	/// <summary>
+	/// A helper class that provides a flood fill algorithm.
+	/// </summary>
+	public static class TilemapFloodFill
+	{
+		private static Grid<bool> activeFillBuffer = new Grid<bool>();
```

Comment:
We are dealing with static state here now due to the static activeFillBuffer field. Remove static from this class and its members and just use a instance of the TilemapFloodFill in FillTilemapTool to solve this.

Reply:
Yeah, you are right. I tought that this class should be static too.. (see TilemapCulling -> https://github.com/AdamsLair/duality/blob/master/Source/Plugins/Tilemaps/Core/Tilemaps/TilemapCulling.cs).

Reply:
I think I can see the arguments on both sides here. It makes sense to keep it static for consistency reasons, but it also makes sense not to, for the reasons @Barsonax mentioned. I'm personally leaning towards the instance variant, now that I've seen it mentioned. 

One request I'd add is to make sure the same instance of the flood fill algorithm is re-usable for multiple operations, re-using its internal buffers along the way to avoid allocations.

Reply:
FillTilemapTool is reused for multiple operations right? Then making the buffer a instance field of FillTilemapTool will make it reuse the buffer for multiple operations. If not then it never reused it in the first place before these changes.

Reply:
Yep, the tools are kept alive and re-used. Even if they weren't though, I think an "algorithm class" instance should be re-usable by design whenever possible.